### DISCUSSION TITLE
[Feature Request] XPath support

### DISCUSSION THREAD

**Discussion by jimustafa** [HDFGroup/hdf5#2894] [Ideas]
**Is your feature request related to a problem? Please describe.**
For files with a complex object hierarchy, it can sometimes be difficult to select a subset of objects based on their path or attributes.

**Describe the solution you'd like**
Given that an H5 file has an object hierarchy, it may be possible to implement functionality to access objects using XPath expressions. This has also been discussed briefly in h5py/h5py#932. Probably a big ask, but likely still worth consideration.

**Describe alternatives you've considered**
None yet. Open to suggestions.

**Additional context**
None.

**Comment by qkoziol** [comment]
Any pointer to XPath, so I could think about this more?

**Reply by jimustafa** [reply]
Some resources that I have used in the past include:
- https://www.w3schools.com/xml/xpath_intro.asp
- https://www.tutorialspoint.com/xpath/
- https://docs.python.org/3/library/xml.etree.elementtree.html#xpath-support

One way to play around with XPath would be to copy some XML from somewhere like https://www.w3schools.com/xml/xpath_examples.asp ([books.xml](https://www.w3schools.com/xml/books.xml)) into an XPath tester, such as http://xpather.com/.
